### PR TITLE
[Tutorial] Update AutoMM Detection Tutorial

### PR DESCRIPTION
**Pull request by FANGAreNotGnu**
- Fixing index to show detection's tutorial correctly.
- Remove unnecessary warnings while calling `setup_save_path` in eval and pred

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/predictor.py [hunk 1]
```diff
@@ -1935,6 +1935,7 @@ def evaluate_coco(
         self._save_path = setup_save_path(
             old_save_path=self._save_path,
             model_loaded=self._model_loaded,
+            warn_if_exist=False,
```
Comment (sxjscience) [inline multimodal/src/autogluon/multimodal/predictor.py:1938]: Shall we just enable this flag in the tutorial mode?

We can detect tutorial mode by detection `AUTOMM_TUTORIAL_MODE` in os.environ: https://github.com/awslabs/autogluon/blob/3f56287b66f8935f4f19450fd08788878bd40b73/multimodal/src/autogluon/multimodal/predictor.py#L387

Comment (FANGAreNotGnu) [reply multimodal/src/autogluon/multimodal/predictor.py:1938]: This should also be False if it's not in tutorial. We'll not create a new path or overwrite anything in `evaluate()` or `predict()`. And thus it's not necessary to give a warning

File: docs/tutorials/multimodal/object_detection/index.rst [hunk 2]
```diff
@@ -30,6 +30,17 @@ Quick Start
       How to train high quality object detection model with MultiModalPredictor in under 5 minutes on COCO format dataset.
 
 
+Quick Start
+------------------
+.. container:: cards
+
+   .. card::
+      :title: Quick Start on a Tiny COCO Format Dataset
```
Comment (yongxinw) [inline docs/tutorials/multimodal/object_detection/index.rst:38]: Is this for the Tiny Motorbike dataset? Do we want to add the dataset name in the title

File: docs/tutorials/multimodal/object_detection/quick_start/index.rst [hunk 3]
```diff
@@ -4,7 +4,7 @@ Quick Start
 
    .. card::
       :title: Quick Start on a Tiny COCO Format Dataset
```
Comment (yongxinw) [inline docs/tutorials/multimodal/object_detection/quick_start/index.rst:6]: Is this for the Tiny Motorbike dataset? Do we want to add the dataset name in the title

**Review by bryanyzhu — APPROVED** [review]
Let's merge this first to get index right, then fix the rest minor things, thank you.

**Review by yongxinw — APPROVED** [review]

**Pull request merged**